lib/kredis/types/flag.rb

```diff
@@ -8,10 +8,18 @@ def mark(expires_in: nil, force: true)
   end
 
   def marked?
-    exists?
+    exists? || exists_after_default_value?
   end
 
   def remove
     del
   end
+
+  private
+    def exists_after_default_value?
+      !!default do |default_value|
+        mark if default_value
+        !!default_value
+      end
```

Comment:
This is very difficult to follow. Need to clarify this code somehow.

Reply:
Refactored